iothub/device/tests/DeviceClientTests.cs

```diff
@@ -425,7 +425,7 @@ public async Task DeviceClient_OnMethodCalled_NullMethodRequest_With_SetMethodHa
 
             bool isMethodHandlerCalled = false;
 #pragma warning disable CS0618 // Type or member is obsolete
-            deviceClient.SetMethodHandler("testMethodName", (payload, context) =>
```

Comment:
I'd recommend doing a sanity check to ensure we don't have an existing test testing the `deviceClient.SetMethodHandlerAsync` method. If we do, then the obsolete test can be simply removed.